### PR TITLE
EFA-392 Simplify MessageViewFragment presentation

### PR DESCRIPTION
**Pull request by ig-garcia**
We stopped using 2 or 3 loaders. Current issue seemed to be gone out of the box, also further features should be easier to implement.

### CODE REVIEW

File: k9mail/src/main/java/com/fsck/k9/extensions/LocalMessage.kt [hunk 1]
```diff
@@ -13,6 +15,16 @@ fun LocalMessage.isMessageIncomplete(): Boolean {
...
 
+fun LocalMessage?.isValidForHandshake() = this != null
+        && (account?.isPlanckPrivacyProtected ?: false)
+        && from != null
+        && from.size == 1
+        && PlanckUtils.isRatingReliable(planckRating)
+        && getRecipients(Message.RecipientType.CC).isNullOrEmpty()
+        && getRecipients(Message.RecipientType.BCC).isNullOrEmpty()
+        && !getRecipients(Message.RecipientType.TO).isNullOrEmpty()
+        && !from.first().address.equals(account.email, ignoreCase = true) // sender not my own account
```
Comment (pzplanck) [inline k9mail/src/main/java/com/fsck/k9/extensions/LocalMessage.kt:26]: could be extracted to the dedicated function

File: k9mail/src/main/java/com/fsck/k9/planck/PlanckProvider.kt [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+    companion object {
+        /**
+         * If is outgoing any copy of the message encrypted (yellow, green, and un secure for some) it will be putted in this position,
+         * if not, all copies will be unencrypted.
+         */
+        const val ENCRYPTED_MESSAGE_POSITION = 0
+        const val PLANCK_OWN_USER_ID = "pEp_own_userId"
+        const val HALF_FINGERPRINT_LENGTH = 24
+
+        //long TIMEOUT = 4 * 60 * 60 * 1000;
```
Comment (pzplanck) [inline k9mail/src/main/java/com/fsck/k9/planck/PlanckProvider.kt:378]: unused commented code line

File: k9mail/src/main/java/com/fsck/k9/planck/PlanckProviderImplKotlin.kt [hunk 3]
```diff
@@ -571,6 +579,63 @@ class PlanckProviderImplKotlin(
...
+            val planckMsg = srcMsg!!
+            planckMsg.dir = Message.Direction.Incoming
+            planckMsg.recvBy = PlanckUtils.createIdentity(Address(account.email), context)
+
+            Timber.d("%s %s", TAG, "decryptMessage() before decrypt")
+            decReturn = engine.get().decrypt_message(srcMsg, Vector(), 0)
+            val decryptReturn = decReturn!!
+            Timber.d("%s %s", TAG, "decryptMessage() after decrypt")
+
+            when (decReturn!!.rating) {
```
Comment (pzplanck) [inline k9mail/src/main/java/com/fsck/k9/planck/PlanckProviderImplKotlin.kt:601]: should we catch the possible exception and throw forward it further too?

Comment (pzplanck) [reply k9mail/src/main/java/com/fsck/k9/planck/PlanckProviderImplKotlin.kt:601]: I see that later we are classifying this case as AppCannotDecryptException

Comment (ig-garcia) [reply k9mail/src/main/java/com/fsck/k9/planck/PlanckProviderImplKotlin.kt:601]: yes, it is just a wrapper, inside a runCatching anything goes more or less.

File: k9mail/src/main/java/com/fsck/k9/extensions/LocalMessage.kt (hunk 1 above)
Comment (ig-garcia) [reply k9mail/src/main/java/com/fsck/k9/extensions/LocalMessage.kt:26]: which part? since this is already a function

Comment (pzplanck) [reply k9mail/src/main/java/com/fsck/k9/extensions/LocalMessage.kt:26]: Line 26

File: k9mail/src/main/java/com/fsck/k9/planck/PlanckProviderImplKotlin.kt [hunk 4]
```diff
@@ -1063,20 +1128,21 @@ class PlanckProviderImplKotlin(
...
-            keys?.forEach { key ->
-      //          identities.add(KeyListItem(key.first, key.second, engine.get().blacklist_is_listed(key.first)))
+    override val blacklistInfo: List<KeyListItem>?
+        @WorkerThread
+        get() = runBlocking(PlanckDispatcher) {
+            try {
+                val identities: MutableList<KeyListItem> = ArrayList()
+                val keys = engine.get().OpenPGP_list_keyinfo("")
+                keys?.forEach { key ->
+                    //          identities.add(KeyListItem(key.first, key.second, engine.get().blacklist_is_listed(key.first)))
```
[the remark this answers is not in the document]
Comment (ig-garcia) [reply k9mail/src/main/java/com/fsck/k9/planck/PlanckProviderImplKotlin.kt:1138]: This did not change for a while, just the black list functionality which is gone from the core now. Somehow still we did not remove it from the app. We may get confirmation and remove it eventually.

[the remark this answers is not in the document]
Comment (pzplanck) [reply k9mail/src/main/java/com/fsck/k9/planck/PlanckProviderImplKotlin.kt:1138]: we better keep track of those deprecated functionalities somewhere and notify PO team, 
while working with the feature include it within the internal configuration or at least in the current class which will allow not only turn on/turn off the feature, but effortlessly remove once requested.

File: k9mail/src/main/java/com/fsck/k9/extensions/LocalMessage.kt (hunk 1 above)
Comment (ig-garcia) [reply k9mail/src/main/java/com/fsck/k9/extensions/LocalMessage.kt:26]: I would not extract that line alone, it is missing context of nulllability etc.

**Pull request merged**